### PR TITLE
Matrix norm for order 1 added.

### LINKED ISSUE #12565: Matrix Norm for order 1 under development
Consider the following piece of code :

```
In [1]: from sympy import Matrix, Symbol, trigsimp, cos, sin, oo

In [2]: from sympy import pprint

In [3]: M= Matrix(3, 3, [1, 3, 0, -2, -6, 0, 3, 9, 6])

In [4]: pprint(M)
⎡1   3   0⎤
⎢         ⎥
⎢-2  -6  0⎥
⎢         ⎥
⎣3   9   6⎦

In [5]: M.norm(1)
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-5-e528283fb202> in <module>()
----> 1 M.norm(1)

/home/mamakancha/sympy/sympy/matrices/matrices.py in norm(self, ord)
   4382 
   4383             else:
-> 4384                 raise NotImplementedError("Matrix Norms under development")
   4385 
   4386     def nullspace(self, simplify=False):

NotImplementedError: Matrix Norms under development

```

### PR DESCRIPTION
**Pull request by mamakancha**
Fixes #12565 

Note : Matrix Norm for order 1 is the maximum sum of column of absolute values in a given matrix

Consider the following code :

```
In [1]: from sympy import Matrix, pprint

In [2]: M= Matrix(3, 3, [1, 3, 0, -2, -1, 0, 3, 9, 6])

In [3]: pprint(M)
⎡1   3   0⎤
⎢         ⎥
⎢-2  -1  0⎥
⎢         ⎥
⎣3   9   6⎦

In [4]: M.norm(1)
Out[4]: 13

```

### CODE REVIEW

**Comment by mamakancha** [comment]
@jksuom Please review

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -4366,7 +4366,22 @@ def norm(self, ord=None):
...
+            if ord == 1: #maximum column sum
+                k = 0
+                max_= 0
+                for i in range(self.shape[1]):
+                    sum_ = 0
+                    a = k
+                    for j in range(self.shape[0]):
+                        sum_ += abs(self[a])
+                        a += self.shape[1]
+                    if(max_ <= sum_):
```
Comment (debugger22) [inline sympy/matrices/matrices.py]:
```
if sum_ > max_:
    max_ = sum_
```
This should be more readable.

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -4366,7 +4366,22 @@ def norm(self, ord=None):
...
+                        max_ = sum_
+                    k+=1
```
Comment (debugger22) [inline sympy/matrices/matrices.py]: Spaces

**Comment by debugger22** [comment]
Tests

File: sympy/matrices/matrices.py [hunk 3]
```diff
@@ -4366,7 +4366,22 @@ def norm(self, ord=None):
 
         # Matrix Norms
         else:
-            if ord == 2:  # Spectral Norm
+            if ord == 1: #maximum column sum
```
Comment (debugger22) [inline sympy/matrices/matrices.py]: `#maximum column sum` - > `# maximum column sum`

**Comment by debugger22** [comment]
Perhaps you can use `map` and `sum` to get rid of loops.

**Comment by debugger22** [comment]
If this is not a gsoc patch requirement, it'd be a good idea to discuss the usability of the feature/function on issue before actually implementing it.

**Comment by mamakancha** [comment]
Yep. It's not a gsoc patch requirement.
Actually I have a course in Numerical Methodology this semester so I tried to use SymPy to solve some questions with matrices. 
The `.norm()` function is available but with very limited functionality in `matrices/matrices.py`
Norms can be of various orders where each order has a specific property. Since the order 1 for norm was not implemented , I tried my hands on it. 
Kindly pardon the late response as I my exams going on :)

**Comment by mamakancha** [comment]
ping @jksuom

**Comment by jksuom** [comment]
Would it be possible to make use of Python built-in functions `sum` and `max`?

**Comment by mamakancha** [comment]
Sure. I can try to work on it. Will get back to you soon.

**Comment by gxyd** [comment]
@mamakancha can you please address the requested changes?

**Comment by gxyd** [comment]
Closing this PR in favor of #12616 .

**Pull request closed without merging**